eduNEXT/eox-theming
# DS-188-eox-theming support for Nutmeg release

## Description

This PR adds eox-theming support for Nutmeg release

## Testing instructions

1. Use this strain.yml file to create a nutmeg project

```bash
STRAIN_NAME: nutmeg
STRAIN_TUTOR_VERSION: v14.0.5
DEV_PROJECT_NAME: nutmeg_dev
LOCAL_PROJECT_NAME: nutmeg_local
CMS_HOST: studio.nutmeg.edunext.link
LMS_HOST: lms.nutmeg.edunext.link
PLATFORM_NAME: Distro nutmeg
STRAIN_TUTOR_PLUGINS:
  - REPO: tutor-contrib-edunext-distro
    VERSION: v3.0.0-rc1
    EGG: tutor-contrib-edunext-distro==v3.0.0-rc1
    PACKAGE_NAME: distro
    PROTOCOL: ssh
STRAIN_EXTRA_COMMANDS:
  - APP: 'tutor'
    COMMAND: 'distro enable-themes'
STRAIN_VOLUME_OVERRIDES:
  edxapp:
    - DESTINATION: ../../src/edxapp/edx-platform
      LOCATION: /openedx/edx-platform
    - DESTINATION: ../../src/edxapp/eox-tenant
      LOCATION: /openedx/extra_deps/eox-tenant
    - DESTINATION: ../../src/edxapp/eox-theming
      LOCATION: /openedx/extra_deps/eox-theming
STRAIN_EXPORT_VOLUMES:
  - CONTAINER: lms
    LOCATION: /openedx/edx-platform
    DESTINATION: src/edxapp/edx-platform
  - CONTAINER: lms
    LOCATION: /openedx/extra_deps/eox-tenant
    DESTINATION: src/edxapp/eox-tenant
  - CONTAINER: lms
    LOCATION: /openedx/extra_deps/eox-theming
    DESTINATION: src/edxapp/eox-theming
DISTRO_EOX_THEMING_DPKG:
  index: git
  name: eox-theming
  private: false
  repo: eox-theming
  domain: github.com
  path: eduNEXT
  protocol: https
  variables:
    development: {}
    production: {}
  version: JDB/nutmeg_release_support
```

2. Run to create project

```bash
stack strain dev configure -k export_vols -k override_vols
stack strain dev init
stack strain dev configure -s export_vols -s override_vols
stack strain dev start
```

3. Use  [ednx-test-themes](https://github.com/eduNEXT/ednx-test-themes) repo to test eox-theming

```bash
cd src/edxapp/edx-platform/themes
git clone git@github.com:eduNEXT/ednx-test-themes.git
```
4. In the file `env/apps/openedx/settings/lms/development.py` you must configure the appropriate path

```bash
COMPREHENSIVE_THEME_DIRS.extend(["/openedx/edx-platform/themes/ednx-test-themes/edx-platform"])
```
5. To test each of the cases, the settings shown must be attached to the settings of the microsite, if you don't have a microsite create in `lms.nutmeg.edunext.link:8000/admin/eox_tenant/microsite/` and adds in subdomain key `lms.nutmeg.edunext.link`

```bash
{
    "EDNX_USE_SIGNAL": true,
    "PLATFORM_NAME": "TEST-THEMING",
    "SITE_NAME": "lms.nutmeg.edunext.link",
    "THEME_OPTIONS": {
        "scripts": {
            "/login": [
                {
                    "content": "alert('This is a test for the inline script');",
                    "type": "inline"
                }
            ]
        },
        "theme": {
            "grandparent": "test-3",
            "name": "test-1",
            "parent": "test-2"
        }
    },
    "course_org_filter": [
        "edX"
    ]
}
```
6. Go to browser `lms.nutmeg.edunext.link:8000` and check the theme, something like https://github.com/eduNEXT/ednx-test-themes/blob/master/captures/1-not-auth.png
7. Go to `/login` and check extra-scripts, something like https://github.com/eduNEXT/ednx-test-themes/blob/master/captures/extra-scripts.png
8. Change microsite settings, theme color have to be diferent. Test steps 6. and 7. again. You find something like https://github.com/eduNEXT/ednx-test-themes/blob/master/captures/2-not-auth.png

```bash
        "theme": {
            "grandparent": "test-1",
            "name": "test-3",
            "parent": "test-2"
        }
```
10. Go to `lms.nutmeg.edunext.link:8000/account/settings` and click on `Reset Your Password`, the file generated inside the lms container, in the path `/tmp/openedx/emails/` 

```bash
tutor dev dc exec lms bash
cd ../../tmp/openedx/emails/
```
 Open the file and ckeck something like https://github.com/eduNEXT/ednx-test-themes/blob/master/captures/1-email.png


## Checklist for Merge
